qiita_db/support_files/patches/python_patches/47.py

```diff
@@ -0,0 +1,680 @@
+
+        # Loop through all the biom tables associated with the current analysis
+        # so we can create the initial set of artifacts
+        for biom_data in analysis_bioms:
+            # Get the path of the BIOM table
+            sql = """SELECT filepath, mountpoint
+                     FROM qiita.filepath
+                        JOIN qiita.data_directory USING (data_directory_id)
+                     WHERE filepath_id = %s"""
+            TRN.add(sql, [biom_data['filepath_id']])
+            # Magic number 0: There is only a single row in the query result
+            fp_info = TRN.execute_fetchindex()[0]
+            filepath = join(db_dir, fp_info['mountpoint'], fp_info['filepath'])
+
+            # We need to check if the BIOM table has been rarefied or not
+            table = load_table(filepath)
+            depths = set(table.sum(axis='sample'))
+            if len(depths) == 1:
+                # The BIOM table was rarefied
+                # Create the initial unrarefied artifact
+                initial_biom_artifact_id = create_non_rarefied_biom_artifact(
+                    analysis, biom_data, table)
+                # Create the rarefaction job
+                rarefaction_job_id, params = create_rarefaction_job(
+                    depths.pop(), initial_biom_artifact_id, analysis,
+                    srare_cmd_id)
+                # Create the rarefied artifact
+                rarefied_biom_artifact_id = create_rarefied_biom_artifact(
+                    analysis, srare_cmd_id, biom_data, params,
+                    initial_biom_artifact_id, rarefaction_job_id,
+                    srare_cmd_out_id)
+            else:
+                # The BIOM table was not rarefied, use current table as initial
+                initial_biom_id = transfer_file_to_artifact()
+
+            # Loop through all the jobs that used this biom table as input
+            sql = """SELECT *
+                     FROM qiita.job
+                     WHERE reverse(split_part(reverse(
+                        options::json->>'--otu_table_fp'), '/', 1)) = %s"""
+            TRN.add(sql, [filepath])
+            analysis_jobs = TRN.execute_fetchindex()
+            for job_data in analysis_jobs:
+                # Identify which command the current job exeucted
+                if job_data['command_id'] == 1:
+                    # Taxa summaries
+                    cmd_id = sum_taxa_cmd_id
+                    params = ('{"biom_table":%d,"metadata_category":"",'
+                              '"sort":false}' % initial_biom_id)
+                    output_artifact_type_id = ts_atype_id
+                    cmd_out_id = sum_taxa_cmd_out_id
+                elif job_data['command_id'] == 2:
+                    # Beta diversity
+                    cmd_id = bdiv_cmd_id
+                    tree_fp = loads(job_data['options'])['--tree_fp']
+                    params = ('{"biom_table":%d,"tree":"%s","metrics":'
```

Comment:
the default bdiv metrics differ -- is it the case that all analyses here have a tree?